com.unity.netcode.gameobjects/Runtime/NetworkVariable/CollectionSerializationUtility.cs

```diff
@@ -0,0 +1,664 @@
+        {
+            reader.ReadByteSafe(out byte full);
+            if (full == 1)
+            {
+                value.Dispose();
+                reader.ReadValueSafe(out value, Allocator.Persistent);
+                return;
+            }
+            ByteUnpacker.ReadValuePacked(reader, out int length);
+            var changes = new ResizableBitVector(Allocator.Temp);
+            using var toDispose = changes;
+            {
+                reader.ReadNetworkSerializableInPlace(ref changes);
+
+                var previousLength = value.Length;
+                if (length != value.Length)
+                {
+                    var newArray = new NativeArray<T>(length, Allocator.Persistent);
+                    unsafe
+                    {
+                        UnsafeUtility.MemCpy(newArray.GetUnsafePtr(), value.GetUnsafePtr(), math.min(newArray.Length * sizeof(T), value.Length * sizeof(T)));
+                    }
+                    value.Dispose();
+                    value = newArray;
+                }
+
+                unsafe
+                {
+                    var ptr = (T*)value.GetUnsafePtr();
+                    for (var i = 0; i < value.Length; ++i)
+                    {
+                        if (changes.IsSet(i))
+                        {
+                            if (i < previousLength)
+                            {
+                                NetworkVariableSerialization<T>.ReadDelta(reader, ref ptr[i]);
+                            }
+                            else
+                            {
+                                NetworkVariableSerialization<T>.Read(reader, ref ptr[i]);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        public static void WriteListDelta<T>(FastBufferWriter writer, ref List<T> value, ref List<T> previousValue)
+        {
+            if (value == null || previousValue == null)
+            {
+                writer.WriteByteSafe(1);
+                NetworkVariableSerialization<List<T>>.Write(writer, ref value);
+                return;
+            }
+            using var changes = new ResizableBitVector(Allocator.Temp);
+            int minLength = math.min(value.Count, previousValue.Count);
+            var numChanges = 0;
+            for (var i = 0; i < minLength; ++i)
+            {
+                var val = value[i];
+                var prevVal = previousValue[i];
+                if (!NetworkVariableSerialization<T>.AreEqual(ref val, ref prevVal))
+                {
+                    ++numChanges;
+                    changes.Set(i);
+                }
+            }
+
+            for (var i = previousValue.Count; i < value.Count; ++i)
+            {
+                ++numChanges;
+                changes.Set(i);
+            }
+
+            if (numChanges >= value.Count * 0.9)
```

Comment:
A brief description about how if the changes are more than 90% of the list then just do a full update.